grouper/public_key.py

```diff
@@ -79,8 +89,55 @@ def delete_public_key(session, user_id, key_id):
         KeyNotFound if specified key wasn't found
```

Comment:
What's wrong with this docstring? That function throws KeyNotFound by virtue of calling get_public_key and not catching KeyNotFound from it.